### DISCUSSION TITLE
Why not having a `get` method on element

### DISCUSSION THREAD

**Discussion by geomagilles** [kwebio/kweb-core#502] [Ideas]
It seems there are only setters on an element. Is there any way to get attributes like classes?

Context: I'm writing a HTML parser - I end up with a correct elements tree, and I would like to add specific behavior like `elt.classes(...)` to update the classes already present in the HTML. The issue is that I need help finding a way to access this info.

**Comment by sanity** [comment]
kweb.html.ElementReader served this purpose however it was deprecated because most of the time the DOM is mirroring server-side state making it unnecessary to read it except for things like `<input>` values for which there is explicit support.

Are you parsing the HTML server-side?

**Reply by geomagilles** [reply]
Indeed. I encounter this need when parsing HTML server-side from a large string to create Kweb elements. Then I want to apply some commands on those elements but I do not know their existing attributes. Those are in the initial string, but I can not access this info easily (see https://github.com/kwebio/kweb-core/discussions/497#discussioncomment-6141441)
